### ISSUE TITLE
[BUG] incorrect typing on get_s3_filesystem has led open_virtual_mfdataset astray

### ISSUE THREAD

**Issue by itcarroll** [nsidc/earthaccess#906]
### Is this issue already tracked somewhere, or is this a new report?

- [X] I've reviewed existing issues and couldn't find a duplicate for this problem.

### Current Behavior

The `earthaccess.open_virtual_mfdataset` function with `access="direct"` returns a `KeyError: 0`.

Here the `earthaccess.get_s3_filesystem` is given `granules[0]` when it should be given `granules`.
https://github.com/nsidc/earthaccess/blob/b9dec8bcbbe4bbdf7eadb318076ee2b2f67e2e69/earthaccess/dmrpp_zarr.py#L95

Here the typing is misleading and should be `Optional[list[DataGranule]]`.
https://github.com/nsidc/earthaccess/blob/b9dec8bcbbe4bbdf7eadb318076ee2b2f67e2e69/earthaccess/api.py#L402

### Expected Behavior

The `earthaccess.open_virtual_mfdataset` function with `access="direct"` returns a dataset.

### Steps To Reproduce

```python
import earthaccess

results = earthaccess.search_data(
    short_name="MUR-JPL-L4-GLOB-v4.1",
    temporal=("2010-01-01", "2010-01-31"),
)
earthaccess.open_virtual_mfdataset(
    results,
    access="direct",
    load=False,
    concat_dim="time",
    coords="all",
    compat="override",
    combine_attrs="drop_conflicts",
)
```

### Environment

```markdown
- OS: openscapes.2i2c
- Python: 3.12.0
```

**Comment by ayushnag** [comment]
@betolink This should be a pretty quick fix. Can I also change [this line](https://github.com/nsidc/earthaccess/blob/b9dec8bcbbe4bbdf7eadb318076ee2b2f67e2e69/earthaccess/api.py#L419) of `get_s3_filesystem` to just use a single granule `result` rather than `results[0]`?

**Comment by betolink** [comment]
@ayushnag Agreed, I think using a list is still the recommended way to go and this is an artifact of some missions having their own endpoints like SWOT and NISAR etc.  @itcarroll is there a difference in `List` vs `list`? which one should we use in this case?

**Comment by itcarroll** [comment]
`typing.List` is a [deprecated alias](https://docs.python.org/3/library/typing.html#typing.List) to `list`. Matter of preference, so I guess I would just be consistent within any given module.

**Comment by danielfromearth** [comment]
I am running into a `KeyError` too (see screenshot). 
It seems like it's probably the same as what's already in this _issue_, but I'm not totally sure... can someone check that it looks the same and that no new issue is necessary?

![temp](https://github.com/user-attachments/assets/04e0fd3b-cda5-4d79-93ed-980f8057e931)

**Comment by mfisher87** [comment]
> is there a difference in `List` vs `list`? which one should we use in this case?

We should use Ruff to upgrade our whole codebase to use builtins now that 3.9 is our oldest supported version. It's in the pyupgrade ruleset (`UP`) if I remember correctly!

**Comment by ayushnag** [comment]
> It seems like it's probably the same as what's already in this issue, but I'm not totally sure... can someone check that it looks the same and that no new issue is necessary?

Yes, this is the same issue. There is a PR to fix it that should be merged soon (https://github.com/nsidc/earthaccess/pull/923)
